database/rrdcalc.c

```diff
@@ -485,18 +509,18 @@ inline RRDCALC *rrdcalc_create_from_template(RRDHOST *host, RRDCALCTEMPLATE *rt,
     rc->update_every = rt->update_every;
     rc->options = rt->options;
 
-    if(rt->exec) rc->exec = strdupz(rt->exec);
-    if(rt->recipient) rc->recipient = strdupz(rt->recipient);
-    if(rt->source) rc->source = strdupz(rt->source);
-    if(rt->units) rc->units = strdupz(rt->units);
+    if(rt->exec) rc->exec = string_strdupz(rt->exec);
+    if(rt->recipient) rc->recipient = string_strdupz(rt->recipient);
+    rc->source = string_dup(rt->source);
+    if(rt->units) rc->units = string_strdupz(rt->units);
     if(rt->info) {
-        rc->info = strdupz(rt->info);
-        rc->original_info = strdupz(rt->info);
+        rc->info = string_strdupz(rt->info);
+        rc->original_info = string_dup(rc->info);
```

Comment:
On this line, instead to duplicate a value from template like original code, you changed for the current alert, please, bring the template back.

Reply:
It is the same. `rc->info` gets the template at line 517, then `rc->original_info` gets it from the `rc->info`.

`string_strdupz()` does an index search.
`string_dup()` just increases the reference counter of the string already in the index. Faster.